### PR TITLE
fix(auth): sends current user with attributes to hub

### PR DESCRIPTION
**Pull request by haverchuck**
#### Description of changes
This addresses an issue in which the `signIn` hub event after `confirmSignIn` invocation contains a `CognitoUser` that does not have attributes. The issue is that we were broadcasting the user _passed_ to the `confirmSignIn` method, which represented the state of the user in the midst of the sign in flow. We solved this by calling `currentUserPoolUser` immediately before dispatching the event and adding the attributes to existing user object (we also pass the updated user to the `resolve` in the auth callback). We then add the missing value to the existing object. We maintain the same object though. 

Finally, there are other APIs  with signIn Hub dispatch sites that could potentially benefit from this fix.

#### Description of how you validated changes
Manual Testing

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/Auth.ts [hunk 1]
```diff
@@ -1182,14 +1186,13 @@ export class AuthClass {
 						} catch (e) {
 							logger.debug('cannot get cognito credentials', e);
 						} finally {
-							that.user = user;
-
+							const currentUser = await this.currentUserPoolUser();
```
Comment (elorzafe) [inline packages/auth/src/Auth.ts:1189]: You might want to wrap that into a `try-catch` in case that method throws?

What do you think?

Comment (haverchuck) [reply packages/auth/src/Auth.ts:1189]: We don't do that in signIn API, but makes sense.

**Review by jimblanc — APPROVED** [review]

**Review by elorzafe — APPROVED** [review]
I am good with this. Only comment is if you can add a unit test where the `this.currentUserPoolUser` fails and you can still see the user on the hub event

File: packages/auth/src/Auth.ts [hunk 2]
```diff
@@ -1183,7 +1183,13 @@ export class AuthClass {
 							logger.debug('cannot get cognito credentials', e);
 						} finally {
 							that.user = user;
-
+							let currentUser;
```
Comment (elorzafe) [inline packages/auth/src/Auth.ts:1186]: Nit: this is not a `const` inside of try-catch block?

**Review by elorzafe — APPROVED** [review]
Looks good to me

Just a minor nit pick comment

**Review by jimblanc — APPROVED** [review]

**Review by elorzafe — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**